### PR TITLE
WebLogic Xmlrules review

### PR DESCRIPTION
**Pull request by mareknovotny**
All outstanding lines assigned to me from google spreadsheet

### CODE REVIEW

**Comment by mareknovotny** [comment]
@lincolnthree or @jsight could you review this PR?

File: rules-reviewed/jboss-eap/weblogic/tests/weblogic-xml-descriptors.windup.test.xml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+   <testDataPath>data</testDataPath>
+   <rulePath>../</rulePath>
+   <ruleset>
+      <rules>
+         <rule id="weblogic-xml-descriptors-01000-test">
+            <when>
+               <not>
+                   <or>
+                       <classification-exists classification="WebLogic EAR Application Descriptor" />    
+                       <!-- hint-exists message="weblogic-application.xml Elements mapping" /-->
```
Comment (lincolnthree) [inline rules-reviewed/jboss-eap/weblogic/tests/weblogic-xml-descriptors.windup.test.xml]: This comment should be removed.

File: rules-reviewed/jboss-eap/weblogic/tests/weblogic-xml-descriptors.windup.test.xml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+               <not>
+                  <classification-exists
+                     classification="WebLogic Web Application Descriptor" />
+               </not>
+            </when>
+            <perform>
+               <fail message="WebLogic Web Application Descriptor not found!" />
+            </perform>
+         </rule>
+         <!--  rule id="weblogic-xml-descriptors-13000-test">
```
Comment (lincolnthree) [inline rules-reviewed/jboss-eap/weblogic/tests/weblogic-xml-descriptors.windup.test.xml]: This comment should be removed?

File: rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+                </iteration>
+            </perform>
+        </rule>
+        <rule id="weblogic-xml-descriptor-03000">
+            <when>
+                <xmlfile matches="//*[local-name()='delay-updates-until-end-of-tx']/text()"/>
+            </when>
+            <perform>
+                <iteration>
+                    <hint title="WebLogic EJB specific configuration" severity="optional" effort="0">
```
Comment (lincolnthree) [inline rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml]: Why are there separate hints here?

File: rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+                <xmlfile as="default" matches="/wl9:weblogic-ejb-jar">
+                    <namespace prefix="wl9" uri="http://www.bea.com/ns/weblogic/90"/>
+                </xmlfile>
+            </when>
+            <perform>
+                <iteration>
+                    <classification title="WebLogic EJB XML" severity="mandatory" effort="3">
+                        <link href="https://access.redhat.com/articles/1326823" description="Map weblogic-ejb-jar.xml Elements to the jboss-ejb3.xml Descriptor" />
+                    </classification>                        
+                    <!-- xslt description="JBoss EJB Descriptor (Windup-Generated)" template="transformations/xslt/weblogic9-ejb-to-jboss.xsl" extension="-jboss.xml"/-->
```
Comment (lincolnthree) [inline rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml]: Should this comment be removed?

File: rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+                <xmlfile as="default" matches="/wl10:weblogic-ejb-jar">
+                    <namespace prefix="wl10" uri="http://www.bea.com/ns/weblogic/10.0"/>
+                </xmlfile>
+            </when>
+            <perform>
+                <iteration>
+                    <classification title="WebLogic EJB XML" severity="mandatory" effort="3">
+                        <link href="https://access.redhat.com/articles/1326823" description="Map weblogic-ejb-jar.xml Elements to the jboss-ejb3.xml Descriptor" />
+                    </classification>                        
+                    <!-- xslt description="JBoss EJB Descriptor (Windup-Generated)" template="transformations/xslt/weblogic10-ejb-to-jboss.xsl" extension="-jboss.xml"/-->
```
Comment (lincolnthree) [inline rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml]: This comment should be removed?

File: rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+            </perform>
+        </rule>
+        <rule id="weblogic-xml-descriptor-17000">
+            <when>
+                <xmlfile as="default" matches="/wl:annotation-manifest | /annotation-manifest">
+                    <namespace prefix="wl" uri="http://www.bea.com/2004/03/wlw/external-config/"/>
+                </xmlfile>
+            </when>
+            <perform>
+                <iteration>
```
Comment (lincolnthree) [inline rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml]: Iteration elements are not required unless you are filtering, selecting variables, or doing some iterating some not iterating conditions in the <perform> clause.

Comment (lincolnthree) [inline rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml]: perform clause*

File: rules-reviewed/jboss-eap/weblogic/tests/weblogic-xml-descriptors.windup.test.xml (hunk 2 above)
Comment (mareknovotny) [inline rules-reviewed/jboss-eap/weblogic/tests/weblogic-xml-descriptors.windup.test.xml]: idk, I haven't prepared test data (therefore it is commented) as this whole rule weblogic-xml-descriptors-13000 is not clear to me. Please see https://github.com/mareknovotny/windup-rulesets/blob/xmlrules-review/rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml#L222

File: rules-reviewed/jboss-eap/weblogic/tests/weblogic-xml-descriptors.windup.test.xml (hunk 1 above)
Comment (mareknovotny) [inline rules-reviewed/jboss-eap/weblogic/tests/weblogic-xml-descriptors.windup.test.xml]: Probably, even the rule adds both classification and hint too, should it be tested both?

File: rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml (hunk 3 above)
Comment (mareknovotny) [inline rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml]: my mistake

File: rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml (hunk 4 above)
Comment (mareknovotny) [inline rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml]: I kept all xslt lines commented to have it for later if we want to update them (xslt for EAP6), but they could be removed. No strong opinion on this.

File: rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml (hunk 5 above)
Comment (mareknovotny) [inline rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml]: I kept all xslt lines commented to have it for later if we want to update them (xslt for EAP6), but they could be removed. No strong opinion on this.

File: rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml (hunk 6 above)
Comment (mareknovotny) [inline rules-reviewed/jboss-eap/weblogic/weblogic-xml-descriptors.windup.xml]: ok, I will remove it.

**Comment by mareknovotny** [comment]
@lincolnthree  I didn't realized that you merged my first commit so I have added new additional commit respecting almost all your comments, now it seems it is not rebased and in conflict. I think we need to close this PR.

**Pull request closed without merging**